### PR TITLE
changes & additions to elasticache

### PR DESCRIPTION
**Pull request by tsaridas**
Added new functions to module
- create_replication_group ( this one was there by my previous commit but did some changes )
- group_exists
- describe_replication_group 
  Changed the wait parameter to wait for node / group to be available and then return. This can cause long wait times but I saw no reason to wait for creating state and my change is helpful for creating a chain of 
- create master node
- create group with master node as master
- create replica node of group

elasticache-master-node:
  boto_elasticache.present:
    - name: mymasternode
    - num_cache_nodes: 1
    - cache_node_type: cache.t1.micro
    - engine: redis
    - engine_verion: 2.8.6
    - wait: True
    - preferred_availability_zone: us-east-1a

elasticache-group:
  boto_elasticache.creategroup:
    - name: my-group
    - primary_cluster_id: mymasternode
    - replication_group_description: descriptionofmygroup
    - wait: True

elasticache-replica:
  boto_elasticache.present:
    - name: myreplica
    - replication_group_id: my-group
    - preferred_availability_zone: us-east-1c
    - wait: False

Changed that you no longer require any parameter in order to create a node
This helps when you need to create a replica.

Added new functions to states 
- creategroup

Changed that you no longer require any parameters in order to create a group or node.
This helps when you need to create a replica.

I'm open for discussion.
Thanks

### CODE REVIEW

File: salt/modules/boto_elasticache.py [hunk 1]
```diff
@@ -84,7 +84,121 @@ def exists(name, region=None, key=None, keyid=None, profile=None):
...
+                ret[_key] = None
+        if _key == 'pending_modified_values':
+            if val:
+                ret[_key] = val
+            else:
+                ret[_key] = None
+    return ret
+
+
+def get_config(name, region=None, key=None, keyid=None, profile=None):
```
Comment (cachedout) [inline salt/modules/boto_elasticache.py]: Did this argspec change? We don't want to break existing calls to this function.

**Comment by tsaridas** [comment]
@cachedout I'm not sure which one you are referring to but I'm positive I'm not breaking any calls.

**Comment by cachedout** [comment]
Sorry for the confusion. I left a note on the line in question but GitHub can make those tricky to spot. I am referring to ``get_config``function which looks like it no longer contains the ``node_info`` kwarg.

**Comment by tsaridas** [comment]
@cachedout But I added that in a previous commit.

**Comment by thatch45** [comment]
These changes should be safe, thanks @tsaridas !

**Comment by cachedout** [comment]
@tsaridas Ah, I did not know that. That's why I asked. :] Thanks.

**Comment by tsaridas** [comment]
@cachedout no problem :)

**Pull request merged**